app/src/main/java/com/github/bgabriel998/softwaredevproject/GeonamesHandler.java

```diff
@@ -1,99 +1,235 @@
+                            + tagValueFromJsonNotNull("historic", jTags)
+                            + tagValueFromJsonNotNull("landuse", jTags)
+                            + tagValueFromJsonNotNull("leisure", jTags)
+                            + tagValueFromJsonNotNull("natural", jTags)
+                            + tagValueFromJsonNotNull("shop", jTags)
+                            + tagValueFromJsonNotNull("sport", jTags)
+                            + tagValueFromJsonNotNull("tourism", jTags);
+                    //remove first "," (quite ugly, I know)
+                    if (poi.mDescription.length()>0)
+                        poi.mDescription = poi.mDescription.substring(1);
+                    //We could try to replicate Nominatim/lib/lib.php/getClassTypes(), but it sounds crazy for the added value.
+                    poi.mUrl = tagValueFromJson("website", jTags);
+
                 }
-                catch (Exception e){
-                    Log.e("GeonamesHandler", e.toString());
+                if ("node".equals(poi.mCategory)){
+                    poi.mLocation = geoPointFromJson(jo);
                 }
+                if (poi.mLocation != null)
+                    pois.add(poi);
             }
-        });
-        queryThread.start();
+            return pois;
+        } catch (JsonSyntaxException e) {
+            Log.e(BonusPackHelper.LOG_TAG, "OverpassAPIProvider: parsing error.");
+            return null;
+        }
+    }
+
+    /**
+     * Extract value identified by key of JSON
+     * @param key key tag for value extraction
+     * @param jTags JSON object containing tags
+     * @return Value
+     */
+    private String tagValueFromJson(String key, JsonObject jTags){
+        JsonElement jTag = jTags.get(key);
+        if (jTag == null)
+            return null;
+        else
+            return jTag.getAsString();
+    }
+
+    /**
+     * Extract value identified by key of JSON with null handling
+     * @param key key tag for value extraction
+     * @param jTags JSON object containing tags
+     * @return Value
+     */
+    private String tagValueFromJsonNotNull(String key, JsonObject jTags){
+        String v = tagValueFromJson(key, jTags);
+        return (v != null ? ","+v : "");
     }
 
     /**
-     * getSurroundingPeaksResult read the result of the query startGetSurroundingPeaks
-     * @return filtered list of POI, containing only mountain
+     * Create location Geopoint ot of Json
+     * @param jLatLon JSON object containing lat and long
+     * @return Location of POI
      */
-    public ArrayList<POI> getSurroundingPeaksResult() throws InterruptedException {
-        if(queryThread == null)
-            throw new IllegalArgumentException("Can't call getSurroundingPeaksResult,please call startGetSurroundingPeaks first");
-        if(queryThread.isAlive())
-            queryThread.join();
-        return filterPOI(POIs);
+    private GeoPoint geoPointFromJson(JsonObject jLatLon){
```

Comment:
```suggestion
    private GeoPoint getGeoPointFromJson(JsonObject jLatLon){
```

Reply:
This function is also copied from the provider API. I prefer to leave it as it is